src/classes/query.py

```diff
@@ -257,6 +302,48 @@ def relative_path(self):
         # Convert path to relative (based on current working directory of Python)
         return os.path.relpath(file_path, info.CWD)
 
+    def getAudioData(self):
+        # Ensure that UI attribute exists
+        if "ui" not in self.data:
+            self.data["ui"] = {}
+            self.save()
+
+        audio_data = self.data["ui"].get("audio_data", False)
+        if audio_data and len(audio_data) > 1:
+            log.info("Audio Data already retrieved.")
+            return
+        if not audio_data:
+            # Placeholder value. Communicates that data is being retrieved
+            self.data["ui"]["audio_data"] = [-999]
```

Comment:
Let's handle the [-999] only value - and protect against from accidentally generating the audio_data again

Reply:
Possibly return `True` and `False` based on if things are success, or need to be tried again